parquet-hadoop/src/main/java/org/apache/parquet/crypto/AesCipher.java

```diff
@@ -0,0 +1,128 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.crypto;
+
+
+import javax.crypto.Cipher;
+import javax.security.auth.DestroyFailedException;
+
+import org.apache.parquet.ShouldNeverHappenException;
+import org.apache.parquet.crypto.ModuleCipherFactory.ModuleType;
+
+import java.io.IOException;
+import java.security.SecureRandom;
+
+
+public class AesCipher {
+
+  public static final int NONCE_LENGTH = 12;
+  public static final int GCM_TAG_LENGTH = 16;
+
+  static final int AAD_FILE_UNIQUE_LENGTH = 8;
+
+  protected static final int CTR_IV_LENGTH = 16;
+  protected static final int GCM_TAG_LENGTH_BITS = 8 * GCM_TAG_LENGTH;
+  protected static final int CHUNK_LENGTH = 4 * 1024;
+  protected static final int SIZE_LENGTH = ModuleCipherFactory.SIZE_LENGTH;
+
+  protected EncryptionKey aesKey;
+  protected final SecureRandom randomGenerator;
+  protected Cipher cipher;
+  protected final byte[] localNonce;
+  protected boolean wipedOut;
+
+
+  AesCipher(AesMode mode, byte[] keyBytes) throws IllegalArgumentException, IOException {
+    if (null == keyBytes) {
+      throw new IllegalArgumentException("Null key bytes");
+    }
+
+    aesKey = new EncryptionKey(keyBytes);
+    randomGenerator = new SecureRandom();
+    localNonce = new byte[NONCE_LENGTH];
+    wipedOut = false;
+  }
+
+  public static byte[] createModuleAAD(byte[] fileAAD, ModuleType moduleType, 
+      short rowGroupOrdinal, short columnOrdinal, short pageOrdinal) {
+    byte[] typeOrdinalBytes = new byte[1];
+    typeOrdinalBytes[0] = moduleType.getValue();
+    if (ModuleType.Footer == moduleType) {
+      return concatByteArrays(fileAAD, typeOrdinalBytes);      
+    }
+
+    byte[] rowGroupOrdinalBytes = shortToBytesLE(rowGroupOrdinal);
+    byte[] columnOrdinalBytes = shortToBytesLE(columnOrdinal);
+    if (ModuleType.DataPage != moduleType && ModuleType.DataPageHeader != moduleType) {
+      return concatByteArrays(fileAAD, typeOrdinalBytes, rowGroupOrdinalBytes, columnOrdinalBytes); 
+    }
+
+    byte[] pageOrdinalBytes = shortToBytesLE(pageOrdinal);
+    return concatByteArrays(fileAAD, typeOrdinalBytes, rowGroupOrdinalBytes, columnOrdinalBytes, pageOrdinalBytes);
+  }
+
+  public static byte[] createFooterAAD(byte[] aadPrefixBytes) {
+    return createModuleAAD(aadPrefixBytes, ModuleType.Footer, (short) -1, (short) -1, (short) -1);
+  }
+
+  /**
+   * Update last two bytes with new page ordinal (instead of creating new page AAD from scratch)
+   * @param pageAAD
+   * @param newPageOrdinal
+   */
+  public static void quickUpdatePageAAD(byte[] pageAAD, short newPageOrdinal) {
+    byte[] pageOrdinalBytes = shortToBytesLE(newPageOrdinal);
+    int length = pageAAD.length;
+    System.arraycopy(pageOrdinalBytes, 0, pageAAD, length-2, 2);
+  }
+
+  static byte[] concatByteArrays(byte[]... arrays) {
```

Comment:
Add keyword 'private'

Reply:
its called from other classes in this package